recipes/import-assertions-to-attributes/src/workflow.ts

```diff
@@ -0,0 +1,154 @@
+import { styleText } from 'node:util';
+import { api } from '@codemod.com/workflow';
+import type { Api } from '@codemod.com/workflow';
+import type { Edit } from '@ast-grep/napi';
+
+// We run this codemod on all JS/TS files
+// including CJS, MJS, TS, and JSX files.
+const globPattern = '**/*.{cjs,mjs,js,jsx,?(d.)cts,?(d.)mts,?(d.)ts,tsx}';
+
+async function processModule(astGrep: Api['astGrep'], contexts: Api['contexts']) {
+	await astGrep({
+        rule: {
+            any: [
+				/**
+				 * cover `import { something } from './module.json' assert { type: 'json' };`
+				 *
+				 * We are searching for an ERROR child
+				 * because the import assertions syntax isn't correctly parsed
+				 * by the parser, so we need to handle it manually.
+				 */
+				{
+					kind: 'import_statement',
+					has: {
+						kind: 'ERROR',
+						inside: {
+							kind: 'import_statement'
+						}
+					}
+				},
+				// cover `import('./module.json', { assert: { type: 'json' } })`
+				{
+					kind: 'call_expression',
+					pattern: "import($SPECIFIER, { assert: $ASSERT })",
+				}
+            ],
+        },
+    }).replace(({ getNode }) => {
+		const node = getNode();
+		const kind = node.kind();
+		const edits: Array<Edit> = [];
+
+		if (kind === 'import_statement') {
+			// Modify import statement with assert
+			const errorChild = node
+				.children()
+				.find( child => child.kind() === 'ERROR');
+
+			if (!errorChild) {
+				// If no ERROR child, skip this node as it might not contain assert
+				return undefined;
+			}
+
+			// Check if the error child contains "assert" keyword
+			const assertIndex = errorChild
+				.children()
+				.findIndex(child => child.text() === 'assert') + 1;
+
+			if (assertIndex > 0) {
+				const nextChild = errorChild.children()[assertIndex];
+				if (nextChild?.text().trim().startsWith('{')) {
+					// Replace "assert" with "with"
+					edits.push(errorChild.children()[assertIndex - 1].replace('with'));
+				}
```

Comment:
I think this is not how this should be done. I think in the targeting, you can get just "assert" and merely provide its replacement, which will just swap `assert` → `with`. I think once you have the narrow target, this part will just be `node.replace('with')` (where `node` is "assert").